### PR TITLE
[Jecoute] Added surveys management for referents

### PR DESCRIPTION
**Pull request by thomasbisignani**
- Form create / edit
capture d ecran de 2018-10-17 12-04-26

- List view

capture d ecran de 2018-10-16 21-08-04

### CODE REVIEW

File: src/Controller/EnMarche/ReferentController.php [hunk 1]
```diff
@@ -147,6 +153,72 @@ public function committeesAction(CommitteeRepository $committeeRepository, Manag
...
+     */
+    public function jecouteSurveyEditAction(Request $request, Survey $survey): Response
+    {
+        $form = $this
+            ->createForm(SurveyFormType::class, $survey)
+            ->handleRequest($request)
+        ;
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get(SurveyHandler::class)->handle($form->getData());
```
Comment (ottaviano) [inline src/Controller/EnMarche/ReferentController.php]: `$form->getData()` -> `$survey`

File: src/Entity/Jecoute/Survey.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+     */
+    private $name;
+
+    /**
+     * The referent who created the survey.
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Adherent")
+     * @ORM\JoinColumn(onDelete="SET NULL")
+     */
+    private $referent;
```
Comment (ottaviano) [inline src/Entity/Jecoute/Survey.php]: what about `author` instead of `referent`?

Comment (thomasbisignani) [reply src/Entity/Jecoute/Survey.php]: I asked @JulHb , and we can use `creator`, what do you think ?

File: src/Repository/SurveyRepository.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+     * @return Survey[]
+     */
+    public function findAllPublishedByCreator(Adherent $creator): array
+    {
+        $this->checkReferent($creator);
+
+        return $this
+            ->createQueryBuilder('survey')
+            ->addSelect('questions')
+            ->leftJoin('survey.questions', 'questions')
```
Comment (ottaviano) [inline src/Repository/SurveyRepository.php]: innerJoin?

File: src/Jecoute/SurveyQuestionTypeEnum.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    public const SIMPLE_FIELD = 'simple_field';
+
+    public const MULTIPLE_CHOICE_TYPE_LABEL = 'Choix multiples';
+
+    public const UNIQUE_CHOICE_TYPE_LABEL = 'Choix unique';
+
+    public const SIMPLE_FIELD_LABEL = 'Champ libre';
+
+    public const ALL_TYPES = [
```
Comment (ottaviano) [inline src/Jecoute/SurveyQuestionTypeEnum.php]: what about transform this in a method 
```
public static function all(): array
{
  return [
        self::MULTIPLE_CHOICE_TYPE_LABEL => self::MULTIPLE_CHOICE_TYPE,
        self::UNIQUE_CHOICE_TYPE_LABEL => self::UNIQUE_CHOICE_TYPE,
        self::SIMPLE_FIELD_LABEL => self::SIMPLE_FIELD,
  ];
}
```
and we can use it in the FormType with `SurveyQuestionTypeEnum::all()` call and as the callback for `Asset\Choice` `@Asset\Choice(callback={"AppBundle\Jecoute\SurveyQuestionTypeEnum", "all"})`.
and we can remove `getAllTypes` from the Entity :) 

WDYT?

Comment (thomasbisignani) [reply src/Jecoute/SurveyQuestionTypeEnum.php]: Nice idea, thx you :+1:

**Pull request merged**